internal/core/items/flickable.rs

```diff
@@ -241,19 +240,31 @@ impl FlickableData {
                         if crate::animations::current_tick() - pressed_time > DURATION_THRESHOLD {
                             return false;
                         }
-                        let can_move_horiz = (Flickable::FIELD_OFFSETS.viewport
-                            + Empty::FIELD_OFFSETS.width)
-                            .apply_pin(flick)
-                            .get()
-                            > flick.width();
-                        let can_move_vert = (Flickable::FIELD_OFFSETS.viewport
-                            + Empty::FIELD_OFFSETS.height)
-                            .apply_pin(flick)
-                            .get()
-                            > flick.height();
+                        // Check if the mouse was moved more than the DISTANCE_THRESHEOLD in a
+                        // direction in which the flickable can flick
                         let diff = position - inner.pressed_pos;
-                        (can_move_horiz && diff.x.abs() > DISTANCE_THRESHOLD)
-                            || (can_move_vert && diff.y.abs() > DISTANCE_THRESHOLD)
+                        let vp = Flickable::FIELD_OFFSETS.viewport;
+                        let check_horizontal = {
+                            let gap_h = (vp + Empty::FIELD_OFFSETS.width).apply_pin(flick).get()
+                                - flick.width();
+                            gap_h > LogicalLength::default() && {
+                                let x = (vp + Empty::FIELD_OFFSETS.x).apply_pin(flick).get();
+                                (x < LogicalLength::default()
+                                    && diff.x_length() > DISTANCE_THRESHOLD)
+                                    || (x > -gap_h && diff.x_length() < -DISTANCE_THRESHOLD)
+                            }
+                        };
+                        let check_vertical = {
+                            let gap_v = (vp + Empty::FIELD_OFFSETS.height).apply_pin(flick).get()
+                                - flick.height();
+                            gap_v > LogicalLength::default() && {
+                                let y = (vp + Empty::FIELD_OFFSETS.y).apply_pin(flick).get();
+                                (y < LogicalLength::default()
+                                    && diff.y_length() > DISTANCE_THRESHOLD)
+                                    || (y > -gap_v && diff.y_length() < -DISTANCE_THRESHOLD)
+                            }
+                        };
+                        check_horizontal || check_vertical
```

Comment:
👍 if you don't mind that previously there was an evaluation shortcut and now check_vertical is always evaluated.

Reply:
yeah, i think it's fine.  this is not hot code anyway.